src/helpers.cpp

```diff
@@ -508,7 +511,10 @@ ada_really_inline void parse_prepared_path(std::string_view input,
         previous_location = new_location + 1;
         if (path_view == "..") {
           if (!path.empty()) {
-            path.erase(path.rfind('/'));
+            size_t last_delimiter = path.rfind('/');
```

Comment:
we can remove path empty check in here as well.

Reply:
Good catch.